iohk-monitoring/src/Cardano/BM/Backend/ExternalAbstraction.lhs

```diff
@@ -63,9 +62,9 @@ instance Pipe NoPipe where
     write   = \_ _ -> pure ()
     getLine = \_ -> pure ""
 
+#ifndef mingw32_HOST_OS
 instance Pipe UnixNamedPipe where
     data PipeHandler UnixNamedPipe = P Handle
-#ifndef mingw32_HOST_OS
     create pipePath =
         (createNamedPipe pipePath stdFileMode >> (P <$> openFile pipePath ReadWriteMode))
```

Comment:
Why not call `open pipePath` when the pipe exists?

Reply:
good point.
and checking for a pipe would also be useful.